src/libutil/rng.hh

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <random>
+#include <limits>
+
+namespace nix {
+
+// Inspired by the book "A Tour of C++, Third Edition" (ISBN-10 0136816487)
+template<typename T, typename Distribution, typename Engine>
+struct RandomNumberGenerator
```

Comment:
Is this a secure random generator or a pseudo random generator? We should probably put this in the name so people not use it in the wrong place.

Reply:
Maybe mentioning in the header that this should not be used for cryptographic purpose would be a good start.